contrib/cirrus/pr-should-include-tests

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# Intended for use in CI: check git commits, barf if no tests added.
+#
+
+# Docs-only changes are excused
+if [[ "${CIRRUS_CHANGE_TITLE}" =~ CI:DOCS ]]; then
+    exit 0
+fi
+
+# So are PRs where 'No test(s) needed' appears in the Github message
+if [[ "${CIRRUS_CHANGE_MESSAGE,,}" =~ no.tests?.needed ]]; then
+    exit 0
+fi
+
+# HEAD should be good enough, but the CIRRUS envariable allows us to test
+head=${CIRRUS_CHANGE_IN_REPO:-HEAD}
+# Base of this PR. Here we absolutely rely on cirrus.
+base=$(git merge-base ${DEST_BRANCH:-master} $head)
+
+# This gives us a list of files touched in all commits, e.g.
+#    A    foo.c
+#    M    bar.c
+# We look for Added or Modified (not Deleted!) files under 'test'.
+if git diff --name-status $base $head | egrep -q '^[AM]\s+(test/|.*_test\.go)'; then
+    exit 0
+fi
+
+# Nothing changed under test subdirectory.
+#
+# This is OK if the only files being touched are "safe" ones.
+filtered_changes=$(git diff --name-status $base $head |
+                       awk '{print $2}'               |
+                       fgrep -vx .cirrus.yml          |
+                       fgrep -vx go.mod               |
+                       fgrep -vx go.sum               |
+                       egrep -v  '^docs/'             |
+                       egrep -v  '^vendor/')
```

Comment:
Add to this: `version/`, `contrib/`, `hack/` - I need all of those for the release process, and I do not want to have to add tests for new releases.

Reply:
Further, anything that ends in `.md` in the base directory. We should not be requiring this for readme changes.

Reply:
Done; also changelog.txt.